src/analysis/special_functions/integrals.lean

```diff
@@ -90,6 +89,116 @@ by simp only [one_div, integral_inv_of_neg ha hb]
 lemma integral_sin : ∫ x in a..b, sin x = cos a - cos b :=
 by rw integral_deriv_eq_sub' (λ x, -cos x); norm_num [continuous_on_sin]
 
+lemma integral_sin_pow (n : ℕ) : ∫ x in 0..π, sin x ^ (n + 2) =
+  ((n + 1) * ∫ x in 0..π, sin x ^ n) - (n + 1) * ∫ x in 0..π, sin x ^ (n + 2) :=
+begin
+  have h : (λ x, sin x ^ (n + 2)) = λ x, sin x ^ (n + 1) * sin x, { funext, ring },
+  have hv : ∀ x ∈ interval 0 π, has_deriv_at (-cos) (sin x) x,
+  { intros, convert (has_deriv_at_cos x).neg, rw neg_neg },
+  have hu : ∀ x ∈ interval 0 π, has_deriv_at (λ x, sin x ^ (n + 1)) ((n + 1) * cos x * sin x ^ n) x,
+  { intros,
+    convert (has_deriv_at_pow (n + 1) (sin x)).comp x (has_deriv_at_sin x) using 1,
+    simp [mul_right_comm], },
+  have : (λ x, cos x * ((n + 1) * cos x * sin x ^ n)) = λ x, (↑n + 1) * (cos x ^ 2 * sin x ^ n),
+  { funext, ring },
+  conv_lhs { rw h },
+  rw integral_mul_deriv_eq_deriv_mul hu hv _ _,
+  simp only [neg_mul_eq_neg_mul_symm, sin_zero, sin_pi, zero_mul, pi.neg_apply, sub_zero,
+    add_eq_zero_iff, ne.def, zero_add, not_false_iff, one_ne_zero, integral_neg, and_false,
+    zero_pow', sub_neg_eq_add, this, integral_const_mul (↑n + 1)],
+  simp only [cos_square', sub_mul, mul_sub, one_mul, ← pow_add _ 2 n, add_comm 2 n],
+  rw [integral_sub, mul_sub],
+  exact ((continuous_pow n).comp continuous_sin).interval_integrable 0 π,
```

Comment:
Can you go through and add "focusing" braces around sequences of tactics which close a specific goal in tactic proofs when there are multiple goals present? See [the style guide section on tactic proofs](https://leanprover-community.github.io/contribute/style.html#tactic-mode).